### ISSUE TITLE
Add //<summary /> comment to Resources.designer.cs properties.

### ISSUE THREAD

**Issue by dellis1972** [xamarin/xamarin-android#2858]
We should look at adding the following `summary` xml comment to our Resources.designer.cs to provide a nicer IDE experience. It will show the people the actual value of the constant when they hover over it. 

```xml
//<summary>2130772019</summary>
// aapt resource value: 0x7F010033
public const int ActionBarLayout_android_layout_gravity = 2130772019;
```

This suggestion has come from the IDE team.

**Comment by dellis1972** [comment]
@kzu please feel free to add any additional comments/requirements :)

**Comment by kzu** [comment]
yep, the goal here is for users referencing the constant to be able to peek at the underlying value (especially useful for strings) right from the code editor. We could also use this from custom editor completion/quickinfo to show the value even for strings in attributes, say.

**Comment by dellis1972** [comment]
@kzu now that I think about it they will only ever get a `int` value. If you are expecting users to be able to hover over `@string/foo`  and see `"Foo is the value of this string"` we currently cannot do that very easily. The Resource.Designer.cs is a managed reflection of the `R.txt` file or an `R.java` file. These only contains numeric `id` values 

```
int string hello 0x7f05001f
```

So the user would only see `2131034143` . 
I suspect you want to be able to see `"Foo is the value of this string"`.

**Comment by kzu** [comment]
Yes, we'd need the value of the string :(

**Comment by dellis1972** [comment]
@kzu I'll have a think and see if there is a way we can support this, as it would be a nice feature. Just not as easy as I first though. :)

**Comment by kzu** [comment]
Ok, here's another request that is complimentary but more urgent: the summary should contain `[identifier]: [value]`. If we cannot get the `value` right now, at least the `summary` should contain the original identifier. This is important since we can't know right now whether a member name with an underscore (i.e. `app_name`) matches an identifier `app_name` or `app.name` which has had its dot replaced with an underscore to make a valid identifier. This is currently the case OOB for `@style/AppTheme.NoActionBar` which ends up in a member named `AppTheme_NoActionBar`, and another identifier for `@string/app_name`. Right now we cannot tell apart one case from the other to reliably infer the actual indentifier in the resource files.
